tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7860,6 +7860,8 @@ def err_hlsl_missing_dispatchgrid_semantic : Error<
    "Broadcasting node shader '%0' with NodeMaxDispatchGrid attribute must declare an input record containing a field with SV_DispatchGrid semantic">;
 def err_hlsl_dispatchgrid_semantic_already_specified : Error<
    "a field with SV_DispatchGrid has already been specified">;
+def err_hlsl_incompatible_dispatchgrid_semantic_type : Error<
+   "SV_DispatchGrid can only be applied to a declaration of unit/uint16_t scalar, vector, or array up to 3 elements, not %0">;
```

Comment:
It would be better to not list the explicit type since these are type aliases and may end up sugared ore de-sugared during diagnostics or rewriting. `uint` is equivalent to `unsigned int` and `uint16_t` should be equivalent to `unsigned int16_t` (although the later seems to crash DXC 🤕)

```suggestion
   "SV_DispatchGrid can only be applied to a declaration of unsigned 32 or 16-bit integer scalar, vector, or array up to 3 elements, not %0">;
```

Reply:
Updated.